test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -303,3 +307,132 @@ func TestKsctlGeneratedTiers(t *testing.T) {
 		})
 	}
 }
+
+func TestFeatureToggles(t *testing.T) {
+	t.Parallel()
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	base1nsTier, err := hostAwait.WaitForNSTemplateTier(t, "base1ns")
+	require.NoError(t, err)
+
+	t.Run("provision space with enabled feature", func(t *testing.T) {
+		// given
+
+		// Create a new tier which is a copy of base1ns but with an additional ClusterRoleBinding object with "test-feature" annotation.
+		// "feature-test" feature is defined in the ToolchainConfig and has 100 weight
+		tier := tiers.CreateCustomNSTemplateTier(t, hostAwait, "ftier", base1nsTier)
+		createdTier, err := hostAwait.WaitForNSTemplateTier(t, tier.Name)
+		tier = tiers.UpdateCustomNSTemplateTier(t, hostAwait, tier, withClusterRoleBinding(t, createdTier, "test-feature"))
+		require.NoError(t, err)
+
+		// when
+
+		// Now let's create a Space
+		user := NewSignupRequest(awaitilities).
+			Username("featured-user").
+			Email("featured@domain.com").
+			ManuallyApprove().
+			TargetCluster(awaitilities.Member1()).
+			EnsureMUR().
+			SpaceTier("base1ns").
+			TargetCluster(memberAwait).
+			RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...).
+			Execute(t)
+		// and promote that space to the ftier tier
+		tiers.MoveSpaceToTier(t, hostAwait, "featured-user", tier.Name)
+
+		// then
+
+		// Verify that the space has the feature annotation
+		space, err := hostAwait.WaitForSpace(t, user.Space.Name)
+		require.NoError(t, err)
+		require.NotEmpty(t, space.Annotations)
+		assert.Equal(t, "test-feature", space.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey])
+		// and CRB for the that feature has been created
+		crb := &v1.ClusterRoleBinding{}
+		crbName := fmt.Sprintf("%s-%s", user.Space.Name, "test-feature")
+		err = memberAwait.WaitForObject(t, "", crbName, crb, &v1.ClusterRoleBindingList{})
+		require.NoError(t, err)
+
+		t.Run("disable feature", func(t *testing.T) {
+			// when
+
+			// Now let's disable the feature for the Space by removing the feature annotation
+			an := space.Annotations
+			delete(an, toolchainv1alpha1.FeatureToggleNameAnnotationKey)
+			space, err := hostAwait.UpdateSpace(t, user.Space.Name, func(s *toolchainv1alpha1.Space) {
+				s.Annotations = an
```

Comment:
why not
```suggestion
			_, err := hostAwait.UpdateSpace(t, user.Space.Name, func(s *toolchainv1alpha1.Space) {
				delete(space.Annotations, toolchainv1alpha1.FeatureToggleNameAnnotationKey)
```

Reply:
https://github.com/codeready-toolchain/toolchain-e2e/pull/1008/commits/fb48abfece5bc03bcf8d5ac2a5d2e937449d723b